app/views/patients/pledge.js.erb

```diff
@@ -1,24 +1,19 @@
-$('.modal-content').html('')
-$('.modal').attr('id', 'pledge-modal')
-$('.modal-content').append(
-  '<%= escape_javascript(render partial: "patients/pledge", locals: { patient: @patient, disable_next: disable_continue?(@patient) })%>'
+$('.modal-content').html(
+  '<%= escape_javascript(render partial: "patients/pledge",
+                                locals: { patient: @patient,
+                                          disable_next: disable_continue?(@patient) }) %>'
 );
 
-$('.modal').modal('toggle');
-
 <% if (!@patient.pledge_sent) %>
-
-  $('.js-title-step span').slice(1).remove();
```

Comment:
I don't know why this works differently all of a sudden but it does. Moving the slice to be after the modal toggle (it runs as the modal pops up) seems to fix. Not pretty but fine for now, and not worth the time to unravel imo.